apps/prairielearn/src/lib/editors.js

```diff
@@ -82,132 +83,142 @@ export class Editor {
-                // Regardless of whether we error, we'll do a clean and reset:
-                //
-                // If pushing succeeded, the clean will remove any empty directories
-                // that might have been left behind by operations like renames.
-                //
-                // If pushing errored, the reset will get us back to a known good state.
-                await cleanAndResetRepository(this.course, gitEnv, job);
-              }
+        if (!config.fileEditorUseGit) {
+          await syncCourseFromDisk(this.course, startGitHash, job);
+          return;
+        }
 
-              job.data.syncAttempted = true;
+        try {
+          await job.exec('git', ['add', ...this.pathsToAdd], {
+            cwd: this.course.path,
+            env: gitEnv,
+          });
+          await job.exec(
+            'git',
+            [
+              '-c',
+              `user.name="${this.user.name}"`,
+              '-c',
+              `user.email="${this.user.uid}"`,
+              'commit',
+              '-m',
+              this.commitMessage,
+            ],
+            {
+              cwd: this.course.path,
+              env: gitEnv,
+            },
+          );
+        } catch (err) {
+          await cleanAndResetRepository(this.course, gitEnv, job);
+          throw err;
+        }
 
-              await syncCourseFromDisk(this.course, startGitHash, job);
+        try {
+          job.data.pushAttempted = true;
 
-              // As with `job.data.pushSucceeded` above, we'll check this flag
-              // on the `editError` page to know if syncing failed so we can
-              // display appropriate instructions.
-              job.data.syncSucceeded = true;
-            });
+          await job.exec('git', ['push'], {
+            cwd: this.course.path,
+            env: gitEnv,
           });
-        },
-      ],
-      (err) => {
-        callback(err, jobSequenceId);
-      },
-    );
+
+          // We'll look for this flag on the `editError` page to know if
+          // we need to display instructions to recover from a failed push.
+          job.data.pushSucceeded = true;
+        } finally {
+          // Regardless of whether we error, we'll do a clean and reset:
+          //
+          // If pushing succeeded, the clean will remove any empty directories
+          // that might have been left behind by operations like renames.
+          //
+          // If pushing errored, the reset will get us back to a known good state.
+          await cleanAndResetRepository(this.course, gitEnv, job);
+        }
+
+        job.data.syncAttempted = true;
+
+        await syncCourseFromDisk(this.course, startGitHash, job);
+
+        // As with `job.data.pushSucceeded` above, we'll check this flag
+        // on the `editError` page to know if syncing failed so we can
+        // display appropriate instructions.
+        job.data.syncSucceeded = true;
+      });
+    });
+    return jobSequenceId;
```

Comment:
IMO this isn't necessary; the only way the caller will be reliably able to grab this in case of failure is off of `serverJob` returned from `setupEditAsync()`. Remember, if `doEditAsync` errors, it will not return a usable `jobSequenceId`.

Reply:
Agreed. Removed in [ea3d984](https://github.com/PrairieLearn/PrairieLearn/pull/9439/commits/ea3d984b36cb388f772a3f33dcda7bc4224a507c).